src/Simplex/Chat.hs

```diff
@@ -977,92 +978,119 @@ processChatCommand' vr = \case
+                forwardName GroupInfo {groupProfile = GroupProfile {displayName}} = displayName
         CTLocal -> do
-          (CChatItem _ ci) <- withFastStore $ \db -> getLocalChatItem db user fromChatId itemId
-          (mc, _) <- forwardMC ci
-          file <- forwardCryptoFile ci
-          let ciff = forwardCIFF ci Nothing
-          pure (ComposedMessage file Nothing mc, ciff)
+          (errs, items) <- lift $ partitionEithers <$> withStoreBatch (\db -> map (getLocalCI db) (L.toList itemIds))
+          unless (null errs) $ toView $ CRChatErrors (Just user) errs
+          mapM ciComposeMsgReq items
+          where
+            getLocalCI :: DB.Connection -> ChatItemId -> IO (Either ChatError (CChatItem 'CTLocal))
+            getLocalCI db itemId = runExceptT . withExceptT ChatErrorStore $ getLocalChatItem db user fromChatId itemId
+            ciComposeMsgReq :: CChatItem 'CTLocal -> CM ComposeMessageReq
+            ciComposeMsgReq ci = do
+              (mc, _) <- forwardMC ci
+              file <- forwardCryptoFile ci
+              let ciff = forwardCIFF ci Nothing
+              pure (ComposedMessage file Nothing mc, ciff)
         CTContactRequest -> throwChatError $ CECommandError "not supported"
         CTContactConnection -> throwChatError $ CECommandError "not supported"
         where
-          forwardMC :: ChatItem c d -> CM (MsgContent, MsgDirection)
-          forwardMC ChatItem {meta = CIMeta {itemDeleted = Just _}} = throwChatError CEInvalidForward
-          forwardMC ChatItem {content = CISndMsgContent fmc} = pure (fmc, MDSnd)
-          forwardMC ChatItem {content = CIRcvMsgContent fmc} = pure (fmc, MDRcv)
+          forwardMC :: CChatItem c -> CM (MsgContent, MsgDirection)
+          forwardMC (CChatItem _ ChatItem {meta = CIMeta {itemDeleted = Just _}}) = throwChatError CEInvalidForward
+          forwardMC (CChatItem _ ChatItem {content = CISndMsgContent fmc}) = pure (fmc, MDSnd)
+          forwardMC (CChatItem _ ChatItem {content = CIRcvMsgContent fmc}) = pure (fmc, MDRcv)
           forwardMC _ = throwChatError CEInvalidForward
-          forwardCIFF :: ChatItem c d -> Maybe CIForwardedFrom -> Maybe CIForwardedFrom
-          forwardCIFF ChatItem {meta = CIMeta {itemForwarded}} ciff = case itemForwarded of
+          forwardCIFF :: CChatItem c -> Maybe CIForwardedFrom -> Maybe CIForwardedFrom
+          forwardCIFF (CChatItem _ ChatItem {meta = CIMeta {itemForwarded}}) ciff = case itemForwarded of
             Nothing -> ciff
             Just CIFFUnknown -> ciff
             Just prevCIFF -> Just prevCIFF
-          forwardCryptoFile :: ChatItem c d -> CM (Maybe CryptoFile)
-          forwardCryptoFile ChatItem {file = Nothing} = pure Nothing
-          forwardCryptoFile ChatItem {file = Just ciFile} = case ciFile of
-            CIFile {fileName, fileStatus, fileSource = Just fromCF@CryptoFile {filePath}}
-              | ciFileLoaded fileStatus ->
```

Comment:
ui may offer to preemptively download files in case it's possible for any files. if any files are missing it can show a bar near multi select buttons saying "x messages will be sent without files", or alert with confirmation because it doesn't make sense to show for delete actions